<a href="https://colab.research.google.com/github/Emelsmsk/proje/blob/main/proje.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
  for filename in filenames:
    print(os.path.join(dirname, filename))

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
%matplotlib inline

current_directory = os.getcwd()
print("Current working directory:", current_directory)

for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

file_path = os.path.join('/kaggle/input', 'HR_comma_sep.csv')

if os.path.exists(file_path):
    df = pd.read_csv(file_path)
else:
    print(f"Error: File not found at {file_path}")

Current working directory: /content
Error: File not found at /kaggle/input/HR_comma_sep.csv


In [ ]:
import pandas as pd
df = pd.read_csv("HR_comma_sep.csv")

In [ ]:
df

satisfaction_level  last_evaluation  number_project  \
0                    0.38             0.53               2   
1                    0.80             0.86               5   
2                    0.11             0.88               7   
3                    0.72             0.87               5   
4                    0.37             0.52               2   
...                   ...              ...             ...   
14994                0.40             0.57               2   
14995                0.37             0.48               2   
14996                0.37             0.53               2   
14997                0.11             0.96               6   
14998                0.37             0.52               2   

       average_montly_hours  time_spend_company  Work_accident  left  \
0                       157                   3              0     1   
1                       262                   6              0     1   
2                       272                   4              0     1   
3                       223                   5              0     1   
4                       159                   3              0     1   
...                     ...                 ...            ...   ...   
14994                   151                   3              0     1   
14995                   160                   3              0     1   
14996                   143                   3              0     1   
14997                   280                   4              0     1   
14998                   158                   3              0     1   

       promotion_last_5years Department  salary  
0                          0      sales     low  
1                          0      sales  medium  
2                          0      sales  medium  
3                          0      sales     low  
4                          0      sales     low  
...                      ...        ...     ...  
14994                      0    support     low  
14995                      0    support     low  
14996                      0    support     low  
14997                      0    support     low  
14998                      0    support     low  

[14999 rows x 10 columns]

In [ ]:
df.dtypes

satisfaction_level       float64
last_evaluation          float64
number_project             int64
average_montly_hours       int64
time_spend_company         int64
Work_accident              int64
left                       int64
promotion_last_5years      int64
Department                object
salary                    object
dtype: object

In [ ]:
df.columns

Index(['satisfaction_level', 'last_evaluation', 'number_project',
       'average_montly_hours', 'time_spend_company', 'Work_accident', 'left',
       'promotion_last_5years', 'Department', 'salary'],
      dtype='object')

In [ ]:
df.isnull().any()

satisfaction_level       False
last_evaluation          False
number_project           False
average_montly_hours     False
time_spend_company       False
Work_accident            False
left                     False
promotion_last_5years    False
Department               False
salary                   False
dtype: bool

In [ ]:
df = df.rename(columns={'satisfaction_level': 'satisfaction',
                        'last_evaluation': 'evaluation',
                        'number_project': 'projetCount',
                        'average_montly_hours': 'averageMonthlyHours',
                        'time_spend_company': 'yearsAtCompany',
                        'Work_accident': 'workAccident',
                        'promotion_last_5years': 'promotion',
                        'sales' : 'deparment',
                        'left': 'turnover'
                        })

# Ciroyu(turnover)gözden geçirme.

In [ ]:
front = df['turnover']
df.drop(labels=['turnover'], axis=1,inplace = True)
df.insert(0, 'turnover', front)
df.head()

turnover  satisfaction  evaluation  projetCount  averageMonthlyHours  \
0         1          0.38        0.53            2                  157   
1         1          0.80        0.86            5                  262   
2         1          0.11        0.88            7                  272   
3         1          0.72        0.87            5                  223   
4         1          0.37        0.52            2                  159   

   yearsAtCompany  workAccident  promotion Department  salary  
0               3             0          0      sales     low  
1               6             0          0      sales  medium  
2               4             0          0      sales  medium  
3               5             0          0      sales     low  
4               3             0          0      sales     low

Bu kod,bir pandas DataFrame'de('df')yer alan 'turnover' sütununa göre gruplama yaparak,herbir 'turnover'değerinin veri setindeki oranını hesaplar.

In [ ]:
turnover_rate = df.groupby('turnover').size() / len(df)
turnover_rate

turnover
0    0.761917
1    0.238083
dtype: float64

Sütunların temel istatistiksel özet bilgileri.

In [ ]:
df.describe()

turnover  satisfaction    evaluation   projetCount  \
count  14999.000000  14999.000000  14999.000000  14999.000000   
mean       0.238083      0.612834      0.716102      3.803054   
std        0.425924      0.248631      0.171169      1.232592   
min        0.000000      0.090000      0.360000      2.000000   
25%        0.000000      0.440000      0.560000      3.000000   
50%        0.000000      0.640000      0.720000      4.000000   
75%        0.000000      0.820000      0.870000      5.000000   
max        1.000000      1.000000      1.000000      7.000000   

       averageMonthlyHours  yearsAtCompany  workAccident     promotion  
count         14999.000000    14999.000000  14999.000000  14999.000000  
mean            201.050337        3.498233      0.144610      0.021268  
std              49.943099        1.460136      0.351719      0.144281  
min              96.000000        2.000000      0.000000      0.000000  
25%             156.000000        3.000000      0.000000      0.000000  
50%             200.000000        3.000000      0.000000      0.000000  
75%             245.000000        4.000000      0.000000      0.000000  
max             310.000000       10.000000      1.000000      1.000000

Veri kümesi bilgilerine göre yaklaşık 15000 çalışan ve 10 özellik belirtilerek:
1-'turnover':Çalışanların işten ayrılma oranı ortalama %23.81.(en düşük değer=0,en yüksek değer=100(1.0)
2-'satisfaction:Çalışan memnuniyeti ortalama %61.28'dir.(en düşük %36,en yüksek %100)
3-'evaluation:Çalışan değerlendirmeleri ortalama %71.61'dir.(en düşük %36,en yüksek %100)
4-'projectCount':Çalışmada yer alan proje sayısı ortalama 3.80'dir.(en az 2,en çok 7 proje)
5-'averageMonthlyHours':Aylık ortalama çalışma saatleri 201'dir.(en az 96 saat,en çok 310 saat)
'yearsAtCompany':Şirketteki yıllık çalışma süresi ortalama 3.5 yıldır.(en az 2,en çok 10 yıl)
6-'workAccident':İş kazalarının oranı ortalama %14.46'dır.(en düşük %0,en yüksek %100)
7-'promotion:Terfi oranı ortalama %2.13'tür.(en düşük 0,en yüksek %100)

In [ ]:
turnover_Summary = df.groupby('turnover')
turnover_Summary.mean(numeric_only=True)

satisfaction  evaluation  projetCount  averageMonthlyHours  \
turnover                                                               
0             0.666810    0.715473     3.786664           199.060203   
1             0.440098    0.718113     3.855503           207.419210   

          yearsAtCompany  workAccident  promotion  
turnover                                           
0               3.380032      0.175009   0.026251  
1               3.876505      0.047326   0.005321